src/react/experimental/transition-manager.js

```diff
@@ -45,36 +45,42 @@ export default class TransitionManager {
   }
 
   // Process the vewiport change, either ignore or trigger a new transiton.
+  // Return true if a new transition is triggered, false otherwise.
   processViewportChange(nextProps) {
+    let transitionTriggered = false;
 
     // NOTE: Be cautious re-ordering statements in this function.
     if (this._shouldIgnoreViewportChange(nextProps)) {
       this.props = nextProps;
-      return;
+      return transitionTriggered;
     }
 
     const isTransitionInProgress = this._isTransitionInProgress();
+    const currentProps = this.props;
+    this.props = nextProps;
```

Comment:
Why are we setting this here as opposed to at the end? If there is a reason (e.g. other functions using it) that seems subtle and worth a comment (or changing those functions to take a props param).